python/geotag_from_gpx.py

```diff
@@ -109,7 +109,13 @@ def exif_time(filename):
     '''
     m = pyexiv2.ImageMetadata(filename)
     m.read()
-    return m['Exif.Photo.DateTimeOriginal'].value
+    try:
+        image_time = m['Exif.Photo.DateTimeOriginal'].value
+        image_millisec = int(m['Exif.Photo.SubSecTimeOriginal'].value)
+    except ValueError, e:
+        print("Error reading EXIF at {0}: {1}".format(filename, e))
+    image_time = image_time + datetime.timedelta(milliseconds=int(image_millisec))
+    return image_time
 
```

Comment:
Good to add error handling here! Line #117 can still fail if Exif.Photo.SubSecTimeOriginal is missing. Maybe we can set it to be default 0?